diesel/src/expression/functions/mod.rs

```diff
@@ -547,51 +163,9 @@ macro_rules! __diesel_sqlite_register_fn {
 /// # }
 /// ```
 macro_rules! sql_function {
-    ($(#$meta:tt)* fn $fn_name:ident $args:tt $(;)*) => {
-        sql_function!($(#$meta)* fn $fn_name $args -> ());
-    };
-
-    ($(#$meta:tt)* fn $fn_name:ident $args:tt -> $return_type:ty $(;)*) => {
-        sql_function!($(#$meta)* fn $fn_name <> $args -> $return_type);
-    };
-
-    (
-        $(#$meta:tt)*
-        fn $fn_name:ident
-        <
-        $($tokens:tt)*
-    ) => {
-        __diesel_parse_type_args!(
-            data = (
-                meta = ($(#$meta)*),
-                fn_name = $fn_name,
-            ),
-            callback = __diesel_sql_function_body,
-            tokens = ($($tokens)*),
-        );
-    };
-
-    ($fn_name:ident, $struct_name:ident, $args:tt -> $return_type:ty) => {
-        sql_function!($fn_name, $struct_name, $args -> $return_type, "");
-    };
-
-    ($fn_name:ident, $struct_name:ident, $args:tt -> $return_type:ty, $docs:expr) => {
-        sql_function!($fn_name, $struct_name, $args -> $return_type, $docs, "");
-    };
-
-    ($fn_name:ident, $struct_name:ident, ($($arg_name:ident: $arg_type:ty),*)) => {
-        sql_function!($fn_name, $struct_name, ($($arg_name: $arg_type),*) -> ());
-    };
-
-    (
-        $fn_name:ident,
-        $struct_name:ident,
-        $args:tt -> $return_type:ty,
-        $docs:expr,
-        $helper_ty_docs:expr
-    ) => {
-        sql_function_body!($fn_name, $struct_name, $args -> $return_type, $docs, $helper_ty_docs);
-    };
+    ($($args:tt)*) => {
+        sql_function_proc! { $($args)* }
```

Comment:
I'm not sure I follow you. Those re-exports don't show up in docs at all

Reply:
Yes they don't show up, but we might wan't to have them showing up in future because the docs for [custom derives](https://docs.rs/diesel_derives/1.4.0/diesel_derives/derive.Identifiable.html) contain information about the allowed attributes and such things.
Anyway I was wrong here, `sql_function_proc!` needs to be reexported, because otherwise it will not be useable in this macro outside of diesel itself.